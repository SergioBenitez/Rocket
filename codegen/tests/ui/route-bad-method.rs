#![feature(plugin, decl_macro)]
#![plugin(rocket_codegen)]

extern crate rocket;

#[route("GET", "/hello")]
fn get1() -> &'static str { "hi" }

#[route(120, "/hello")]
fn get2() -> &'static str { "hi" }

#[route("CONNECT", "/hello")]
fn get3() -> &'static str { "hi" }
