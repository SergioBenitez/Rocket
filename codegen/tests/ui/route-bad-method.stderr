error: not a string literal
 --> $DIR/route-bad-method.rs:9:9
  |
9 | #[route(120, "/hello")]
  |         ^^^
  |
  = help: valid methods are: `GET`, `PUT`, `POST`, `DELETE`, `HEAD`, `PATCH`, `OPTIONS`

error: 'CONNECT' is not a valid method
  --> $DIR/route-bad-method.rs:12:9
   |
12 | #[route("CONNECT", "/hello")]
   |         ^^^^^^^^^
   |
   = help: valid methods are: `GET`, `PUT`, `POST`, `DELETE`, `HEAD`, `PATCH`, `OPTIONS`

error: aborting due to 2 previous errors

